src/Apps/Order/Routes/__tests__/Respond.test.tsx

```diff
@@ -210,6 +211,25 @@ describe("Offer InitialMutation", () => {
       )
     })
 
+    it("Logging an error with an unrecognized action", () => {
+      const component = getWrapper()
+      const declineRadio = component.find(BorderedRadio).last()
+
+      declineRadio
+        .props()
+        .onSelect({ selected: true, value: "UNRECOGNIZED VALUE" })
+
+      component
+        .find(Button)
+        .last()
+        .props()
+        .onClick({})
+
+      expect(logger.error).toHaveBeenCalledWith(
```

Comment:
I think that this test should verify that an unrecognized action doesn't modify the order, instead of verifying that a message is logged.

Reply:
That's a good point, but we won't have that test available, really, until the counter offer flow is finished since none of the other actions actually modify the order.

Reply:
Though part of my response there is assuming that there isn't a straightforward way to test if the order is modified or not at this point, and instead would have to ensure that `commitMutation` isn't called

Reply:
Curious for your thoughts here @pepopowitz

Reply:
I was thinking about the tests for reject-offer that Ash and I wrote in Exchange when I made this comment. In that case we made sure that the status of the mocked order was still "submitted". After reading your comment, I realize that Reaction is a purely front-end application so my suggestion doesn't make sense.

However, I still think that there is a better way to test this than checking the logger. Is there an error message that would be displayed to the user when an unrecognized action is attempted?

Reply:
I think the assertion that we log an error has value. It says "we will find out when users find an unexpected path through our code."

I also think it makes sense to assert that nothing was changed on the offer, but as you say @williardx, I think the best way to do that in this type of test is to assert that `commitMutation` wasn't called. 

My vote would be "let's have this one test assert that `commitMutation` wasn't called but a logger was". Verification that nothing was changed on the order itself would be great as an end-to-end test, but I don't think that fits here. (Which raises the question - do we have a place for end-to-end tests in this app?)

Reply:
@pepopowitz we do not have "end-to-end" tests in the app! 😞 The closest we currently have is the router tests, which visit a route and _can_ assert what's shown on the page.

This case seems rare (is it even possible?) so it seems like we have a few options:
- Show a generic error modal (we can test that this appears, as we do elsewhere)
- Default to an action like "accept" (since it just kicks them to the review screen, there's not much danger here)

Reply:
Hm yeah, I'm not sure how this case would happen in practice at the moment. Given that, just going to the accept screen seems like a very simple fallback. If we need better error handling around this case we can always add it later when we know we need it.

Reply:
Thanks for explaining that to me. What do you guys think about including the action in the logged message and asserting that `UNRECOGNIZED VALUE` is included?

Reply:
We absolutely should do what you're describing @williardx but this conversation made me think more about why this impossible "default" logic branch exists. I think it's because we're using an open-ended string instead of something like a string enum.

This seems like the kind of thing that would be solved by "making impossible states impossible" with a string enum (or something...I'm no TypeScript expert). Then we wouldn't need a default case, because all actually-valid cases would have been covered. 

No action required by this comment....I'm just thinking out loud.

Reply:
Er, disregard my comment above. I missed this line, where we ARE limiting the valid values - 

 `responseOption: "ACCEPT" | "COUNTER" | "DECLINE" | null`

Maybe we just want to remove the `null` option & `default` case, since we think that's impossible?

Reply:
That makes a lot of sense to me! Feels like a very clean and simple approach 💯